### PR TITLE
DOC: Clean up examples of low-level random access

### PR DESCRIPTION
**Pull request by bashtage**
Test examples on Linux and Windows
Correct bug in Cython example
Improve building instructions for numba examples

### CODE REVIEW

**Comment by bashtage** [comment]
@mattip I don't quite see how the function in distributions.c can be used in Cython code. Am I missing something?

**Comment by mattip** [comment]
You are correct. We [started that discussion](https://github.com/numpy/numpy/issues/14778#issuecomment-554505244) in gh-14778, but did not resolve how to do this cleanly. I am also concerned that exposing [all of the functions](https://github.com/numpy/numpy/blob/master/numpy/random/_generator.pyx#L54) in lines 54-138  increases the C-API we then commit to maintain.

File: numpy/random/_examples/cython/setup.py [hunk 1]
```diff
@@ -11,18 +11,17 @@
 from setuptools.extension import Extension
 from os.path import join, abspath, dirname
 
-curpath = abspath(dirname(__file__))
+path = abspath(dirname(__file__))
```
Comment (mattip) [inline numpy/random/_examples/cython/setup.py]: Why this change? `path` is, I think, too generic.

File: numpy/random/_examples/numba/extending_distributions.py [hunk 2]
```diff
@@ -35,19 +41,19 @@
...
-random_gauss_zig = lib.random_gauss_zig
+random_standard_normal = lib.random_standard_normal
 
 
 def normals(n, bit_generator):
     out = np.empty(n)
     for i in range(n):
-        out[i] = random_gauss_zig(bit_generator)
+        out[i] = random_standard_normal(bit_generator)
     return out
```
Comment (mattip) [inline numpy/random/_examples/numba/extending_distributions.py]: I wonder if we could run this as part of the tests, like we run the `extending.py` example

File: numpy/random/_examples/cython/setup.py (hunk 1 above)
Comment (bashtage) [reply numpy/random/_examples/cython/setup.py]: 7 letters made the line spill below. Excessive beautification.

File: numpy/random/_examples/numba/extending_distributions.py (hunk 2 above)
Comment (bashtage) [reply numpy/random/_examples/numba/extending_distributions.py]: Depends on numba or at least CFFI, so probably not worth it.  But definitely a risk that examples get stale, so ideally these things get run somewhere.

**Comment by bashtage** [comment]
> You are correct. We [started that discussion](https://github.com/numpy/numpy/issues/14778#issuecomment-554505244) in [gh-14778](https://github.com/numpy/numpy/issues/14778), but did not resolve how to do this cleanly. I am also concerned that exposing [all of the functions](https://github.com/numpy/numpy/blob/master/numpy/random/_generator.pyx#L54) in lines 54-138 increases the C-API we then commit to maintain.

I'll move it over there then.
